File: longest_prefix_suffix.cpp
class Solution{
  public:		
	int lps(string s) {
	    int s_len=s.length();
	    int lps[s_len]={0};
	    int start=0;
	   
	    int i=1;
	    while(i<s.length()){
            if(s[start]==s[i]){
                lps[i]=start+1;
                start++;
                i++;
            }
            else{
                if(start>0){
                        
                        start=lps[start-1];
                        // [start]=
                    }
                else{
                    i++;
                }
            }
        }
	    return lps[s_len -1];
	    
	}
};

File: longest_prefix_suffix.js
class Solution {
    lps(s){ 
        //code here
        let lps= new Array();
        let i=0,j=1;
        let sLen=s.length;
        lps[0]=0;
        while(j<sLen){
            if(s[j]==s[i]){
                lps[j]=i+1;
                i++;
                j++;
            }
            else if(i==0){
                // j--;
                lps[j]=0;
                j++;
            }
            else{
                i=lps[i-1];
            }
        }
      
        return lps[sLen-1];
        
    }
}